### PR TITLE
bugfix: make the daemon exit happy

### PR DESCRIPTION
**Pull request by fuweid**
When the daemon server fails, the process will be hang because the
no-one closes the channel. We should exit.

Signed-off-by: Wei Fu <fhfuwei@163.com>

### Ⅰ. Describe what this PR did

Make the daemon exit happy.

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Describe how you did it

Basically, we should run the daemon main in new goroutine and wait for it to exit. Based on this, we can receive the signal and clean it up. 

### Ⅳ. Describe how to verify it

Use available port and it should be success.

```
pouchd --listen tcp://localhost:5000 --debug
```

Use used port and it should be fail

```
python -m SimpleHTTPServer 5000 &
pouchd --listen tcp://localhost:5000 --debug
```

### Ⅴ. Special notes for reviews

@allencloud , @HusterWan and @rudyfly 

Should we need to clean it up like `sigHandle` when we receive error from `Run()`?

cc @shaloulcy

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -181,37 +181,40 @@ func runDaemon() error {
...
-	return err
+		for _, handle := range sigHandles {
+			if err := handle(); err != nil {
+				logrus.Errorf("failed to handle signal: %v", err)
+			}
+		}
+
+		os.Exit(1)
+	case err := <-errCh:
+		// FIXME: should we do the cleanup like signal handle?
```
Comment (HusterWan) [inline main.go]: I think we also should do cleanup when daemon exited.  
I do not know if this case will happen:  pouchd panic and return error, then we should make the daemon exit gracefully. WDYT ?

**Comment by rudyfly** [comment]
LGTM, and add TODO when server cause error, what to do cleanup.

File: main.go (hunk 1 above)
Comment (fuweid) [reply main.go]: Agree. But for now, we should hold the scope like @rudyfly said. Refactor the code in the following PR. WDYT?

Comment (fuweid) [reply main.go]: Confirm with @HusterWan , let us add TODO here.

**Pull request merged**